public/app/features/dashboard-scene/settings/GeneralSettings.tsx

```diff
@@ -22,11 +32,160 @@ export class GeneralSettingsEditView
     const dashboard = getDashboardSceneFor(model);
     const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
+    const { title, description, tags, meta, editable, timepicker, timezone, weekStart, liveNow } = dashboard.useState();
+
+    const onTitleChange = (value: string) => {
+      dashboard.setState({ title: value });
+    };
+
+    const onDescriptionChange = (value: string) => {
+      dashboard.setState({ description: value });
+    };
+
+    const onTagsChange = (value: string[]) => {
+      dashboard.setState({ tags: value });
+    };
+
+    const onFolderChange = (newUID: string, newTitle: string) => {
+      const newMeta = {
+        ...meta,
+        folderUid: newUID || meta.folderUid,
+        folderTitle: newTitle || meta.folderTitle,
+        hasUnsavedFolderChange: true,
+      };
+
+      dashboard.setState({ meta: newMeta });
+    };
+
+    const onEditableChange = (value: boolean) => {
+      dashboard.setState({ editable: value });
+    };
+
+    const onTimeZoneChange = (value: TimeZone) => {
+      dashboard.setState({ timezone: value });
+    };
+
+    const onWeekStartChange = (value: string) => {
+      dashboard.setState({ weekStart: value });
+    };
+
+    const onRefreshIntervalChange = (value: string[]) => {
+      dashboard.setState({
+        timepicker: {
+          ...timepicker,
+          refresh_intervals: value,
+        },
+      });
+    };
+
+    const onNowDelayChange = (value: string) => {
+      dashboard.setState({
+        timepicker: {
+          ...timepicker,
+          nowDelay: value,
+        },
+      });
+    };
+
+    const onHideTimePickerChange = (value: boolean) => {
+      dashboard.setState({
+        timepicker: {
+          ...timepicker,
+          hidden: value,
+        },
+      });
+    };
+
+    const onLiveNowChange = (value: boolean) => {
+      dashboard.setState({ liveNow: value });
+    };
 
     return (
       <Page navModel={navModel} pageNav={pageNav} layout={PageLayoutType.Standard}>
         <NavToolbarActions dashboard={dashboard} />
-        <div>General todo</div>
+        <div style={{ maxWidth: '600px' }}>
+          <Box marginBottom={5}>
+            <Field
+              label={
+                <HorizontalGroup justify="space-between">
+                  <Label htmlFor="title-input">
+                    <Trans i18nKey="dashboard-settings.general.title-label">Title</Trans>
+                  </Label>
+                  {/* TODO: Make the component use persisted model */}
+                  {/* {config.featureToggles.dashgpt && (
+                  <GenAIDashTitleButton onGenerate={onTitleChange} dashboard={dashboard} />
+                )} */}
+                </HorizontalGroup>
+              }
+            >
+              <Input
+                id="title-input"
+                name="title"
+                value={title}
+                onChange={(e: ChangeEvent<HTMLInputElement>) => onTitleChange(e.target.value)}
```

Comment:
Nit: Think this should be updated on blur to avoid too eager set state calls. Similar other text fields that do not require changes to be reflected/visible right away